### PR TITLE
[1813] Readme - Iterate explanation of factory usage

### PR DESCRIPTION
**Pull request by timabell**
Having followed the path, hopefully this is more usage focussed now.
Further iteration welcome.

preview https://github.com/DFE-Digital/manage-courses-backend/blob/1813-readme-factory-usage/spec/README.md

### CODE REVIEW

File: spec/README.md [hunk 1]
```diff
@@ -1,64 +1,88 @@
...
 
-# Describe vs Context vs Feature vs Scenario
+The way we do it:
 
-# Using 'let' and 'begin'
+1. Created objects should be valid by default, so ensure that required
+   associations are built/created.
+2. Avoid building/creating associations that aren't necessary. (i.e. nullable
+   foreign keys, and many-to-many relationships with join tables).
+4. Provide a way to overide associated objects, for has_* relationships this
```
Comment (faissalMT) [inline spec/README.md]: This sequence jumps from 2 to 4 here.

File: spec/README.md [hunk 2]
```diff
@@ -1,64 +1,88 @@
...
+   will need to build associated objects by default, and add them to the parent
+   object in an `after_create` hook.
+5. Provide a way to specify exactly what associated object(s) override
```
Comment (faissalMT) [inline spec/README.md]:
```suggestion
5. Provide a way to specify exactly what associated object(s) override.
```

Comment (faissalMT) [reply spec/README.md]: I will concede that this is very much a nitpick.

File: spec/README.md (hunk 1 above)
Comment (timabell) [reply spec/README.md]: good spot, cheers. done

**Pull request merged**